src/test/java/org/opensearch/security/dlic/dlsfls/CCReplicationTest.java

```diff
@@ -0,0 +1,230 @@
+package org.opensearch.security.dlic.dlsfls;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Supplier;
+import org.opensearch.OpenSearchSecurityException;
+import org.opensearch.cluster.health.ClusterHealthStatus;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.junit.Assert;
+import org.junit.Test;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.action.ActionResponse;
+import org.opensearch.action.ActionType;
+import org.opensearch.action.IndicesRequest;
+import org.opensearch.action.IndicesRequest.Replaceable;
+import org.opensearch.action.admin.cluster.health.ClusterHealthRequest;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.HandledTransportAction;
+import org.opensearch.action.support.IndicesOptions;
+import org.opensearch.action.support.WriteRequest.RefreshPolicy;
+import org.opensearch.action.support.master.AcknowledgedRequest;
+import org.opensearch.action.support.master.AcknowledgedResponse;
+import org.opensearch.client.Client;
+import org.opensearch.client.transport.TransportClient;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.env.Environment;
+import org.opensearch.env.NodeEnvironment;
+import org.opensearch.node.Node;
+import org.opensearch.node.PluginAwareNode;
+import org.opensearch.plugins.ActionPlugin;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.script.ScriptService;
+import org.opensearch.security.OpenSearchSecurityPlugin;
+import org.opensearch.security.support.ConfigConstants;
+import org.opensearch.security.test.DynamicSecurityConfig;
+import org.opensearch.tasks.Task;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.Netty4Plugin;
+import org.opensearch.transport.TransportService;
+import org.opensearch.watcher.ResourceWatcherService;
+
+public class CCReplicationTest extends AbstractDlsFlsTest {
```

Comment:
How this test relates to Cross Cluster replication? Other than action name that can be also `foo/plugins/replication/bar`, I don't see anything specific for replication.

Reply:
The intent is to simulate replication actions as part of this test. As the code is from different plugin altogether, we will not be able to directly access the replication related actions. 
Let me know, If you have any other way to achieve this.